### PR TITLE
add move joint demo

### PR DESCRIPTION
**Pull request by azompas**

### CODE REVIEW

File: sr_ethercat_hand_config/demos/demo_move_joint.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+#
+# You should have received a copy of the GNU General Public License along
+# with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import rospy
+from copy import deepcopy
+from sr_robot_commander.sr_hand_commander import SrHandCommander
+
+
+rospy.init_node("right_hand_demo", anonymous=True)
```
Comment (mikramarc) [inline sr_ethercat_hand_config/demos/demo_move_joint.py]: please move to main

File: sr_ethercat_hand_config/demos/demo_move_joint.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        if joint_number_int == 0:
+            joint_1 = "rh_" + finger + "J1"
+            joint_2 = "rh_" + finger + "J2"
+        else:
+            joint = "rh_" + finger + "J" + joint_number_str
+        flex_degrees = None
+        flex_degrees = raw_input("Select flex position in degrees (default 180): ")
+        try:
+            if flex_degrees:
+                type(float(flex_degrees))
```
Comment (mikramarc) [inline sr_ethercat_hand_config/demos/demo_move_joint.py]: this isnt doing anything, is it?

File: sr_ethercat_hand_config/demos/demo_move_joint.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    rospy.sleep(1.2)
+    while True:
+        joints = list()
+        finger, flex, extend, joint_number_str, joint_number_int = select_finger()
+        if joint_number_int == 0:
+            joint_1 = "rh_" + finger + "J1"
+            joint_2 = "rh_" + finger + "J2"
+        else:
+            joint = "rh_" + finger + "J" + joint_number_str
+        flex_degrees = None
```
Comment (mikramarc) [inline sr_ethercat_hand_config/demos/demo_move_joint.py]: is this necessary?

File: sr_ethercat_hand_config/demos/demo_move_joint.py [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+                if joint_number_int == 0:
+                    flex[joint_1] = 90
+                    flex[joint_2] = 90
+                else:
+                    flex[joint] = 90
+        except ValueError:
+            rospy.logerr("You didn't give a valid value")
+            continue
+
+        ext_degrees = None
```
Comment (mikramarc) [inline sr_ethercat_hand_config/demos/demo_move_joint.py]: necessary?

File: sr_ethercat_hand_config/demos/demo_move_joint.py [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+        ext_degrees = raw_input("Select extension position in degrees (default 0): ")
+        try:
+            if ext_degrees:
+                type(float(ext_degrees))
```
Comment (mikramarc) [inline sr_ethercat_hand_config/demos/demo_move_joint.py]: not doing anything again?

File: sr_ethercat_hand_config/demos/demo_move_joint.py [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+                if joint_number_int == 0:
+                    extend[joint_1] = 0
+                    extend[joint_2] = 0
+                else:
+                    extend[joint] = 0
+        except ValueError:
+            rospy.logerr("You didn't give a valid value")
+            continue
+
+        time_raw = None
```
Comment (mikramarc) [inline sr_ethercat_hand_config/demos/demo_move_joint.py]: same

File: sr_ethercat_hand_config/demos/demo_move_joint.py [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+        time_raw = raw_input("Select the time for each movement in seconds (default 1.0): ")
+        try:
+            if time_raw:
+                type(float(time_raw))
```
Comment (mikramarc) [inline sr_ethercat_hand_config/demos/demo_move_joint.py]: unused type

**Review by mikramarc — CHANGES_REQUESTED** [review]

File: sr_ethercat_hand_config/demos/demo_move_joint.py (hunk 7 above)
Comment (azompas) [reply sr_ethercat_hand_config/demos/demo_move_joint.py]: ensures that a float is given

File: sr_ethercat_hand_config/demos/demo_move_joint.py (hunk 5 above)
Comment (azompas) [reply sr_ethercat_hand_config/demos/demo_move_joint.py]: ensures that a float is given

File: sr_ethercat_hand_config/demos/demo_move_joint.py (hunk 2 above)
Comment (azompas) [reply sr_ethercat_hand_config/demos/demo_move_joint.py]: ensures that a float is given

**Review by mikramarc — APPROVED** [review]

**Pull request merged**